src/App.tsx

```diff
@@ -1,24 +1,316 @@
+      });
+  }
+
+  useEffect(loadTodos, []);
+  // #endregion
+
+  // #region add, delete, update
+  const addTodo = ({ userId, title, completed }: Todo) => {
+    setIsLoading(true);
+    setIsFocused(false);
+
+    const promise = todoService.createTodo({
+      userId, title: title.trim(), completed,
+    })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setNewTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        clearError();
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+        setIsFocused(true);
+      });
+
+    setTempTodo({
+      id: 0, userId: USER_ID, title, completed,
+    });
+
+    return promise;
+  };
+
+  const deleteTodo = (id: number) => {
+    setIsLoading(true);
+    setSelectedId([id]);
+    setIsFocused(false);
+
+    return todoService.deleteTodo(id)
+      .then(() => {
+        setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setSelectedId(null);
+        setIsFocused(true);
+      });
+  };
+
+  const deleteCompleted = () => {
+    setIsLoading(true);
+    setIsFocused(false);
+    setSelectedId(idsOfCompletedTodos);
+
+    return Promise.all(
+      idsOfCompletedTodos.map(id => todoService.deleteTodo(id)),
+    )
+      .then(() => {
+        setTodos(
+          currentTodos => currentTodos.filter(
+            todo => !idsOfCompletedTodos.includes(todo.id),
+          ),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setTodos(
+          currentTodos => currentTodos.filter(
+            todo => !idsOfCompletedTodos.includes(todo.id),
+          ),
+        );
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setSelectedId(null);
+        setIsFocused(true);
+      });
+  };
+
+  const toggleStatus = (todo: Todo) => {
+    const {
+      id, userId, title, completed,
+    } = todo;
+
+    setSelectedId([id]);
+    setIsLoading(true);
+
+    return todoService.updateTodo({
+      id, userId, title, completed: !completed,
+    })
+      .then((newTodo) => {
+        setTodos(currentTodos => {
+          const newTodos = [...currentTodos];
+          const index = newTodos.findIndex(item => item.id === id);
+
+          newTodos.splice(index, 1, newTodo);
+
+          return newTodos;
+        });
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      })
+      .finally(() => {
+        setSelectedId(null);
+        setIsLoading(false);
+      });
+  };
+
+  const toggleAll = () => {
+    setIsLoading(true);
+    const activeTodo = todos.filter(todo => !todo.completed);
+
+    let updatingTodo = [...todos];
+
+    if (activeTodo.length > 0) {
```

Comment:
```suggestion
    if (!!activeTodo.length) {
```

Reply:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/122940921/3d85e131-fe55-473b-9cce-79814483058a)
i catch this error with using !!activeTodo.length